src/ScrollingContainer.js

```diff
@@ -270,15 +266,41 @@ ScrollingContainer.prototype.initScrolling = function () {
         container.position[direction] = Math.round(Position[direction]);
 
         self.updateScrollBars();
-
     };
 
-    this._checkLimit = function (limitName) {
-        if (!(limitName in limitsReached)) {
-            limitsReached[limitName] = true;
-            self.emit(limitName + "Overflow");
+    this._checkLimits = function (direction) {
+        if (!this.scrolling && Math.round(Speed[direction]) === 0) {
+            return;
         }
-    };
+
+        var bounds = this.getInnerBounds();
+
+        var maxScrollDistance = (direction == "y") ?
+            Math.round(Math.min(0, this._height - bounds.height)) :
+            Math.round(Math.min(0, this._width - bounds.width));
+
+        var minReached = targetPosition[direction] >= 0;
+        var maxReached = targetPosition[direction] <= maxScrollDistance;
+
+        if (minReached) {
+            self.emit("min" + direction.toUpperCase() + "Overflow");
+        }
+        if (maxReached) {
+            self.emit("max" + direction.toUpperCase() + "Overflow");
+        }
+
+        if ((minReached || maxReached) && self._boundsChanged(bounds)) {
+            self._checkLimits(direction);
```

Comment:
If the user is reacting to the events sent and it is adding new content to the scrolling area.
This ensures 2 things:
- The bounds are forced to be updated
- It keeps rechecking the limits until the bounds do not change this is useful to prevent a little bouncing back effect if the user tries to scroll too fast.